lib/utils.js

```diff
@@ -349,8 +349,41 @@ async function printLibimobiledeviceInfo () {
   }
 }
 
+/**
+ * Get the IDs of processes listening on the particular system port.
+ * It is also possible to apply additional filtering based on the
+ * process command line.
+ *
+ * @param {string|number} port - The port number.
+ * @param {?Function} filteringFunc - Optional lambda function, which
+ *                                    receives command line string of the particular process
+ *                                    listening on given port, and is expected to return
+ *                                    either true or false to include/exclude the corresponding PID
+ *                                    from the resulting array.
+ * @returns {Array<string>} - the list of matched process ids.
+ */
+async function getPIDsListeningOnPort (port, filteringFunc = null) {
+  const result = [];
+  try {
+    // This only works since Mac OS X El Capitan
+    const {stdout} = await exec('lsof', ['-ti', `tcp:${port}`]);
+    result.push(...(stdout.trim().split(/\n+/)));
+  } catch (e) {
+    return result;
```

Comment:
Should the error at least be logged?

Reply:
it returns non-zero if no processes are listening on the port, so I decided to keep it as is not to create unnecessary confusion.

Reply:
Ok, that makes sense.